File: packages/sdk-python/examples/hello.py
"""Hello-world demo: send a trace using the SDK.

Run after `make dev` is up, ingest is bound to :4318, and workers are
consuming the stream. Then:

    uv run --package judge python packages/sdk-python/examples/hello.py

Refresh http://localhost:3000 — the trace should appear within a few seconds.
"""

from __future__ import annotations

import time

import judge

judge.init(
    api_key="local-dev",
    endpoint="http://localhost:4318",
    project="demo",
)


@judge.trace(name="hello_world")
def answer(query: str) -> str:
    with judge.span("retrieve") as s:
        s.log(query=query, retrieved=["doc-1", "doc-2"])
        time.sleep(0.05)
    with judge.span("generate") as s:
        s.log(model="claude-sonnet-4-6", prompt_chars=len(query))
        time.sleep(0.1)
    return f"answer-to-{query}"


def main() -> None:
    out = answer("what is the capital of France?")
    print(f"sdk -> {out}")
    print("trace pushed. visit http://localhost:3000/")


if __name__ == "__main__":
    main()

File: packages/sdk-ts/examples/hello.ts
/**
 * Hello-world demo: send a trace using the TypeScript SDK.
 *
 * Run after `make dev` is up, ingest is bound to :4318:
 *
 *     pnpm --filter @llm-judge/sdk build
 *     node --experimental-strip-types packages/sdk-ts/examples/hello.ts
 *
 * Or with tsx:
 *     pnpm dlx tsx packages/sdk-ts/examples/hello.ts
 */

import { init, span, trace } from '../src/index.js';

init({
  apiKey: 'local-dev',
  endpoint: 'http://localhost:4318',
  project: 'demo',
});

const answer = trace('hello_world_ts', async (query: string) => {
  const s1 = span('retrieve', { query });
  s1.log({ retrieved: ['doc-1', 'doc-2'] });
  await new Promise((r) => setTimeout(r, 50));
  s1.end();

  const s2 = span('generate', { model: 'claude-sonnet-4-6' });
  await new Promise((r) => setTimeout(r, 100));
  s2.end();

  return `answer-to-${query}`;
});

answer('what is the capital of France?').then((out) => {
  console.log(`sdk(ts) -> ${out}`);
  console.log('trace pushed. visit http://localhost:3000/');
});
